src/pool/srv_pool.c

```diff
@@ -1207,30 +1190,55 @@ pool_svc_map_dist_cb(struct ds_rsvc *rsvc)
 	struct rdb_tx		tx;
 	struct pool_buf	       *map_buf = NULL;
 	uint32_t		map_version;
+	daos_prop_t	       *prop = NULL;
+	uint64_t		prop_bits;
 	int			rc;
 
 	/* Read the pool map into map_buf and map_version. */
 	rc = rdb_tx_begin(rsvc->s_db, rsvc->s_term, &tx);
 	if (rc != 0)
 		goto out;
 	ABT_rwlock_rdlock(svc->ps_lock);
+
 	rc = read_map_buf(&tx, &svc->ps_root, &map_buf, &map_version);
-	ABT_rwlock_unlock(svc->ps_lock);
-	rdb_tx_end(&tx);
 	if (rc != 0) {
 		D_ERROR(DF_UUID": failed to read pool map buffer: %d\n",
 			DP_UUID(svc->ps_uuid), rc);
-		goto out;
+		goto unlock;
 	}
 
-	rc = ds_pool_iv_map_update(svc->ps_pool, map_buf, map_version);
+	prop_bits = DAOS_PO_QUERY_PROP_ALL;
+	rc = pool_prop_read(&tx, svc, prop_bits, &prop);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": read prop failed for pool, rc="DF_RC"\n",
+			DP_UUID(svc->ps_uuid), DP_RC(rc));
+		goto unlock;
+	}
+
+unlock:
+	ABT_rwlock_unlock(svc->ps_lock);
+	rdb_tx_end(&tx);
 	if (rc != 0)
+		goto out;
+
+	rc = ds_pool_iv_map_update(svc->ps_pool, map_buf, map_version);
```

Comment:
It is a bit wired to do IV update in map dist cb.  Hmm, if iv_update is asynchronous(lazy update), probably we do not need this dist_cb (by condition) to do pool map IV update in step_up_cb(), i.e. we can call ds_pool_iv_map_update() directly in step_up_cb()?   @liw what is the intention for broadcast pool map by dist_cb() ?